opentelemetry-jaeger/src/exporter/runtime.rs

```diff
@@ -71,23 +77,13 @@ impl JaegerTraceRuntime for opentelemetry::runtime::AsyncStd {
     }
 }
 
-const INADDR_ANY: &str = "0.0.0.0:0";
-const IN6ADDR_ANY: &str = "[::]:0";
 /// Sample the first address provided to designate which IP family to bind the socket to.
 /// Returns either INADDR_ANY or IN6ADDR_ANY
-fn bind_addr<'a, T: ToSocketAddrs>(sockaddrs: &T) -> &'a str {
-    let sockaddrs = sockaddrs.to_socket_addrs();
-    if sockaddrs.is_err() {
-        return INADDR_ANY;
-    }
-    sockaddrs.unwrap().next().map_or_else(
-        || INADDR_ANY,
-        |s| {
-            if s.is_ipv4() {
-                INADDR_ANY
-            } else {
-                IN6ADDR_ANY
-            }
-        },
-    )
+fn addrs_and_family(host_port: &impl ToSocketAddrs) -> Result<(Vec<SocketAddr>, SocketAddr), io::Error> {
+    let addrs = host_port.to_socket_addrs()?.collect::<Vec<_>>();
+    let family = match addrs.first() {
+        Some(SocketAddr::V4(_)) | None => SocketAddr::from((Ipv4Addr::new(0, 0, 0, 0), 0)),
```

Comment:
I guess we could use `Ipv4Addr::UNSPECIFIED` here instead? And same for `Ipv6Addr`, I suppose.